### DISCUSSION TITLE
Dashboard cannot be made public if it contains a text parameter

### DISCUSSION THREAD

**Discussion by raine** [getredash/redash#6785] [Ideas]
Hey there,

Our use case:

In a kibana dashboard, we have dynamic links to a redash dashboard that contains a UUID as a parameter.

The UUID is used in `WHERE` clause of queries of the dashboard.

The idea is that you can jump from Kibana to a debugging dashboard in Redash.

Problem: The dashboard cannot be made public company-wide because it contains a text based parameter.

![image](https://github.com/getredash/redash/assets/11027/b62d6029-fae6-4548-afbd-d66c7c8405c6)

Our redash instance is behind google workspace login.

My current workaround is to invite team members manually to whom the dashboard is most relevant but in company of over hundred engineers this is not the best solution.

Potential solutions:

- Allow overriding this "security" protection somehow.
- Add a new UUID type for query parameters so that freeform text type would not have to be used. UUID would not allow sql injection which presumably is the threat here.

**Comment by gaecoli** [comment]
This is a security strategy to prevent external attacks.

**Reply by raine** [reply]
Yes, I understand. Any thoughts on suggestions provided?

**Comment by eradman** [comment]
The ability to share dashboards with text parameters from the Web UI is a much needed feature.

One solution might be to require a regex for the value of each text query parameter
```
parameter: volume
match: [-A-Za-z0-9]+
...
```

**Reply by gaecoli** [reply]
Is there any detailed information I'm interested in solving this problem.

**Comment by eradman** [comment]
I think the solution is to add one more input type, currently we have the following

```py
        validators = {
            "text": lambda value: isinstance(value, str),
            "number": _is_number,
            "enum": lambda value: _is_value_within_options(value, enum_options, allow_multiple_values),
            "query": lambda value: _is_value_within_options(
                value,
                [v["value"] for v in dropdown_values(query_id, self.org)],
                allow_multiple_values,
            ),
            "date": _is_date,
            "datetime-local": _is_date,
            "datetime-with-seconds": _is_date,
            "date-range": _is_date_range,
            "datetime-range": _is_date_range,
            "datetime-range-with-seconds": _is_date_range,
        }
```

We could create a new type called `text-pattern` that validates input based on a regex. Only `text` is considered unsafe

```py
    @property
    def is_safe(self):
        text_parameters = [param for param in self.schema if param["type"] == "text"]
        return not any(text_parameters)
```

**Reply by gaecoli** [reply]
Good idea. I try to implement it.

**Reply by eradman** [reply]
@ezraodio1 submitted a solution to this in https://github.com/getredash/redash/pull/7025
